src/clients/HttpTunneler.cc

```diff
@@ -130,6 +130,9 @@ Http::Tunneler::writeRequest()
     HttpHeader hdr_out(hoRequest);
     Http::StateFlags flags;
     flags.peering = true;
+
+    Must(request->masterXaction);
+    request->masterXaction->generatingConnect = true;
```

Comment:
Please make this true/false bracketing logic exception-safe. AFAICT, you can probably bracket just the httpBuildRequestHeader() call. Since C++ does not have `std::scope_exit` yet, we can probably just add a try/catch here.

Reply:
commit a5ad56e.

I include the MemBuf.append operations too, it is not really needed but the code looks better.

I used the HttpTunneler::handleException method which is currently unsused (why this method added?).

Reply:
Addressed in branch commit 693ec2e.

> I used the HttpTunneler::handleException method which is currently unsused (why this method added?).

Difficult to say, but it is possible that the author used the wrong method name: The correct method name would be callException(). handleException() is a part of Ipc::Forwarder API, not AsyncJob API. This is a kind of problem that `override` in C++11 was meant to reduce but HttpTunneler does not use that keyword.

This wrong-method bug is not related to this issue though -- we do not handle exceptions in this issue. See 693ec2e for details.